kadai3-2.html

```diff
@@ -0,0 +1,136 @@
+    margin-right: 10px;
+    font-weight: bold;
+    font-size: large;
+}
+</style>
+
+<body>
+    <h1>ToDoリスト</h1>
+    <input type="radio" name="status" 　value="all" checked>すべて
+    <input type="radio" name="status" 　value="working">作業中
+    <input type="radio" name="status" 　value="done">完了
+    <table id="tasklist">
+        <tr>
+            <th>ID</th>
+            <th>コメント</th>
+            <th>状態</th>
+        </tr>
+    </table>
+    <h2>新規タスクの追加</h2>
+    <p><input type="text" id="taskname"><input type="button" id="addButton" class="button" value="追加"></input></p>
+</body>
+<script>
+const INITIAL_ROW_NUM = 1;
+const DELETE_BUTTON = '削除';
+const STATUS_WORKING = '作業中';
+const table = document.getElementById('tasklist');
+
+document.getElementById('addButton').addEventListener('click', () => {
+    //入力値を取得
+    const todo = {
+        task: document.getElementById('taskname').value,
+        status: STATUS_WORKING
+    }
+
+    //ToDoリストの作成
+    const todos = createTodos(todo);
+
+    //テーブルのリセット
+    while (table.rows.length > INITIAL_ROW_NUM) table.deleteRow(1);
+
+    //ToDoリストの表示
+    displayTodos(todos);
+});
+
+function createTodos(newTodo) {
+    let todos = [];
+    const TODO_COLUMN = 1;
+    const STATUS_COLUMN = 2;
+
+    if (table.rows.length === INITIAL_ROW_NUM) {
+        todos.push(newTodo)
+    } else {
+        for (let i = 1; i < table.rows.length; i++) {
+            const todo = {
+                task: table.rows[i].cells[TODO_COLUMN].innerHTML,
+                status: table.rows[i].cells[STATUS_COLUMN].childNodes[0].value
+            }
+            todos.push(todo);
+        }
+        todos.push(newTodo)
+    }
+
+    return todos;
+}
+
+function displayTodos(todoList) {
+    for (todo of todoList) {
+        //タスク行の追加
+        const fragment = document.createDocumentFragment();
+        const tr = document.createElement('tr');
+        fragment.appendChild(tr);
+
+        //IDカラムの設定
+        const tdId = document.createElement('td');
+        const rowNum = tasklist.rows.length;
+        tdId.textContent = rowNum;
+        tr.appendChild(tdId);
+
+        //コメントカラムの設定
+        const tdComment = document.createElement('td');
+        tdComment.textContent = todo.task;
+        tr.appendChild(tdComment);
+
+        //状態カラムの設定
+        const tdStatus = document.createElement('td');
+        const buttonStatus = document.createElement('input');
+        buttonStatus.setAttribute('type', 'button');
+        buttonStatus.setAttribute('value', todo.status);
+        buttonStatus.setAttribute('class', 'button');
+        tdStatus.appendChild(buttonStatus);
+        tr.appendChild(tdStatus);
+
+        //削除ボタンの設定
+        const buttonDelete = document.createElement('input');
+        buttonDelete.setAttribute('type', 'button');
+        buttonDelete.setAttribute('value', DELETE_BUTTON);
+        buttonDelete.setAttribute('class', 'button');
+        buttonDelete.setAttribute('onclick', 'deleteRow(this)');
```

Comment:
確かにこのような形でもボタンにイベント処理を紐づけることはできますがあまり良い方法ではありません。
（例えば `deleteRow` の名前を変更したくなった時に IDE やエディタの機能を使った安全な名称変更ができない、関数/メソッドの利用箇所を調べようとしたときに調査対象から漏れる等）

以下のように `addEventListener` を利用してみてはいかがでしょうか？

```suggestion
        buttonDelete.addEventListener('click', () => {
            // deleteRow と同じ内容を記述する
        });
```